streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/StreamsMetricsImpl.java

```diff
@@ -284,6 +288,7 @@ public Map<String, String> clientLevelTagMap() {
         final Map<String, String> tagMap = new LinkedHashMap<>();
         tagMap.put(CLIENT_ID_TAG, clientId);
         tagMap.put(PROCESS_ID_TAG, processId);
+        tagMap.put(APPLICATION_ID_TAG, applicationId);
```

Comment:
Pardon me. I see this new tag applied al all metrics, yet the KIP only specifies the `client-state` metric. Am I misunderstanding the scope? thanks!

Reply:
As discussed on the mailing list, we are cleaning this up. Cf https://github.com/apache/kafka/pull/20883